libethereum/Account.cpp

```diff
@@ -34,18 +34,24 @@ using namespace dev::eth::validation;
 
 namespace fs = boost::filesystem;
 
-void Account::setCode(bytes&& _code)
+void Account::setCode(bytes&& _code, u256 const& _version)
 {
-    m_codeCache = std::move(_code);
-    m_hasNewCode = true;
-    m_codeHash = sha3(m_codeCache);
+    auto const newHash = sha3(_code);
+    if (newHash != m_codeHash)
```

Comment:
I think we should have done this check before, too, because without it in case init code returns empty code, `setCode({})` raises `m_hasNewCode` flag, and this makes `commit` function to save into database the pair `EmptySHA3 => ""`. This should better not happen.

Here it also helps with the same problem in case of empty init code. In this case `setCode` is still called to set the version.

Reply:
Separate PR?

Reply:
Yeah can be separate